### PR TITLE
Add start stop commands

### PR DESCRIPTION
**Pull request by ktsakalozos**
Fixes: https://github.com/ubuntu/microk8s/issues/162

### CODE REVIEW

File: microk8s-resources/wrappers/microk8s-start.wrapper [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+set -eu
+
+export PATH="$SNAP/usr/sbin:$SNAP/usr/bin:$SNAP/sbin:$SNAP/bin:$PATH"
+
+sudo snap start ${SNAP_NAME} --enable
```
Comment (kwmonroe) [inline microk8s-resources/wrappers/microk8s-start.wrapper]: Can we check the return code here and potentially offer help?  Something like,
```bash
if rc is 0
  if lock exists
    remove lock
  fi
else
  ## maybe pkill or otherwise stop any half-started services?
  echo 'Failed to start microk8s. See snap logs or journalctl or ...'
  exit 1
fi
```
I know we're `set -e`, so a failed `snap start` would exit before the lock removal, but I think self-documenting would be useful here.

File: microk8s-resources/wrappers/microk8s-stop.wrapper [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+set -eu
+
+export PATH="$SNAP/usr/sbin:$SNAP/usr/bin:$SNAP/sbin:$SNAP/bin:$PATH"
+
+sudo snap stop ${SNAP_NAME} --disable
```
Comment (kwmonroe) [inline microk8s-resources/wrappers/microk8s-stop.wrapper]: As with `start`, can we point the user somewhere useful if `stop` returns non-zero?

File: microk8s-resources/wrappers/run-with-config-args [hunk 3]
```diff
@@ -42,6 +42,12 @@ then
       sleep 6
     done
   fi
+  if [ -e ${SNAP_DATA}/var/lock/stopped.lock ]
+  then
+    # Mark the api server as starting. This is needed incase you
+    # microk8s.stop and then snap start microk8s
+    rm -rf ${SNAP_DATA}/var/lock/stopped.lock &> /dev/null
```
Comment (kwmonroe) [inline microk8s-resources/wrappers/run-with-config-args]: Nice corner case handling, but the `-r` is unnecessary.

File: microk8s-resources/actions/common/utils.sh [hunk 4]
```diff
@@ -1,5 +1,14 @@
 #!/usr/bin/env bash
 
+exit_if_stopped() {
+  # test if the snap is marked as stopped
+  if [ -e ${SNAP_DATA}/var/lock/stopped.lock ]
+  then
+    echo "microk8s is not running, try microk8s.start"
+    exit 1
+  fi
+}
```
Comment (kwmonroe) [inline microk8s-resources/actions/common/utils.sh]: @ktsakalozos while we're in here, what do you think about trying to address a bit of #174 in this function?  I was thinking we expand it like this:
```bash
if -e stopped.lock
    echo "microk8s is not running, try microk8s.start"
    exit 1
else
  if check_apiserver: ## apiserver port unreachable; systemclt status apiserver != running; apiserver process not running; etc:
    while retry < 5
      echo "waiting for microk8s api server to start"
      sleep 2
      if check_apiserver == good
        return
      else
        retry++
    done
    echo "microk8s api server did not start, see journalctl"
    exit 1
  fi
fi
```
Since this PR adds the `exit_if_stopped` check to all the relevant wrappers, i think it makes sense to use this util as a "stopped or not running" gate.

Comment (ktsakalozos) [reply microk8s-resources/actions/common/utils.sh]: I am already on it @kwmonroe but not in this PR. Have a look here https://github.com/ubuntu/microk8s/pull/178/files#diff-0f01327987334a699c82fa97496f4895R10 . With the `microk8s.status` command you will be able to get the status (running or not), and also wait for microk8s to get ready (with a timeout or wait forever). :)

File: microk8s-resources/wrappers/microk8s-stop.wrapper (hunk 2 above)
Comment (ktsakalozos) [reply microk8s-resources/wrappers/microk8s-stop.wrapper]: Done
